storage/rocksdb/CMakeLists.txt

```diff
@@ -161,32 +167,37 @@ ENDIF()
 IF (WITH_JEMALLOC)
   ADD_DEFINITIONS(-DROCKSDB_JEMALLOC)
 ENDIF()
-
-IF (ROCKSDB_CUSTOM_NAMESPACE)
-  ADD_DEFINITIONS(-DROCKSDB_NAMESPACE=${ROCKSDB_CUSTOM_NAMESPACE})
-  ADD_DEFINITIONS(-DROCKSDB_CUSTOM_NAMESPACE=${ROCKSDB_CUSTOM_NAMESPACE})
+IF (WITH_NUMA)
+  ADD_DEFINITIONS(-DNUMA)
 ENDIF()
 
-include(compiler_features)
+IF (CMAKE_SYSTEM_NAME STREQUAL "Linux")
+  ADD_DEFINITIONS(-DOS_LINUX)
+ELSEIF(CMAKE_SYSTEM_NAME MATCHES "Darwin")
+  ADD_DEFINITIONS(-DOS_MACOSX)
+ENDIF()
 
-IF (NOT HAVE_THREAD_LOCAL)
-  MESSAGE(FATAL_ERROR "No '__thread' support found. Not building MyRocks")
+IF (CMAKE_COMPILER_IS_GNUCXX)
+  ADD_COMPILE_OPTIONS(-fno-builtin-memcmp)
 ENDIF()
 
-IF (CMAKE_SYSTEM_PROCESSOR STREQUAL "x86_64")
-  IF (HAVE_SSE42)
-    MESSAGE(STATUS "MyRocks: SSE42 support found")
-    add_compile_flags(${ROCKSDB_ROOT}/util/crc32c.cc COMPILE_FLAGS "-msse4.2") # use Fast_CRC32
-    IF (HAVE_PCLMUL)
-      add_compile_flags(${ROCKSDB_ROOT}/util/crc32c.cc COMPILE_FLAGS "-mpclmul") # use crc32c_3way instead of Fast_CRC32
-    ENDIF()
-  ELSE()
-    IF (ALLOW_NO_SSE42)
-      MESSAGE(WARNING "No SSE42 support found and ALLOW_NO_SSE42 specified, building MyRocks but without SSE42/FastCRC32 support")
+
+INCLUDE(compiler_features)
+
+IF (CMAKE_HOST_SYSTEM_PROCESSOR STREQUAL "x86_64")
+  IF (NOT ROCKSDB_BUILD_ARCH)
+    IF (ROCKSDB_DISABLE_MARCH_NATIVE)
+      # Intel Westmere is oldest CPU that supports SSE 4.2 and PCLMUL instruction sets
+      SET(ROCKSDB_BUILD_ARCH "westmere")
     ELSE()
-      MESSAGE(FATAL_ERROR "No SSE42 support found. Not building MyRocks")
+      SET(ROCKSDB_BUILD_ARCH "native")
     ENDIF()
   ENDIF()
+
+  # CMAKE_CXX_FLAGS are used by CHECK_CXX_SOURCE_RUNS called from ROCKSDB_SET_DEFINTIONS()
+  SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -march=${ROCKSDB_BUILD_ARCH}")
+  MESSAGE(STATUS "MyRocks build architecture: -march=${ROCKSDB_BUILD_ARCH}")
+  ROCKSDB_SET_X86_DEFINTIONS()
 ELSE() # aarch64
```

Comment:
Shouldn't we do something similar to the `AARCH64` branch as well?
I believe `ROCKSDB_DISABLE_MARCH_NATIVE` can be reused here as well
and minimal `ROCKSDB_BUILD_ARCH` can be set to, say, `arvv8-a`

Reply:
done